In [97]:
import pandas as pd

Pandas kütüphanesini import ettikten sonra kaggle'dan indirdiğimiz imdb top 1000 film veri setini dataframe değişkeninin içine tanımlayalım.

In [98]:
df = pd.read_csv("imdb_top_1000.csv")

Dataframe'in ilk 3 satırını görüntüleme:

In [99]:
df.head(3)

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   
2  https://m.media-amazon.com/images/M/MV5BMTMxNT...   

               Series_Title Released_Year Certificate  Runtime  \
0  The Shawshank Redemption          1994           A  142 min   
1             The Godfather          1972           A  175 min   
2           The Dark Knight          2008          UA  152 min   

                  Genre  IMDB_Rating  \
0                 Drama          9.3   
1          Crime, Drama          9.2   
2  Action, Crime, Drama          9.0   

                                            Overview  Meta_score  \
0  Two imprisoned men bond over a number of years...        80.0   
1  An organized crime dynasty's aging patriarch t...       100.0   
2  When the menace known as the Joker wreaks havo...        84.0   

               Director           Star1           Star2          Star3  \
0        Frank Darabont     Tim Robbins  Morgan Freeman     Bob Gunton   
1  Francis Ford Coppola   Marlon Brando       Al Pacino     James Caan   
2     Christopher Nolan  Christian Bale    Heath Ledger  Aaron Eckhart   

            Star4  No_of_Votes        Gross  
0  William Sadler      2343110   28,341,469  
1    Diane Keaton      1620367  134,966,411  
2   Michael Caine      2303232  534,858,444

Poster_Link Sutununu silelim:

In [100]:
df_new=df.drop('Poster_Link',axis=1)
df_new.head()

Series_Title Released_Year Certificate  Runtime  \
0  The Shawshank Redemption          1994           A  142 min   
1             The Godfather          1972           A  175 min   
2           The Dark Knight          2008          UA  152 min   
3    The Godfather: Part II          1974           A  202 min   
4              12 Angry Men          1957           U   96 min   

                  Genre  IMDB_Rating  \
0                 Drama          9.3   
1          Crime, Drama          9.2   
2  Action, Crime, Drama          9.0   
3          Crime, Drama          9.0   
4          Crime, Drama          9.0   

                                            Overview  Meta_score  \
0  Two imprisoned men bond over a number of years...        80.0   
1  An organized crime dynasty's aging patriarch t...       100.0   
2  When the menace known as the Joker wreaks havo...        84.0   
3  The early life and career of Vito Corleone in ...        90.0   
4  A jury holdout attempts to prevent a miscarria...        96.0   

               Director           Star1           Star2          Star3  \
0        Frank Darabont     Tim Robbins  Morgan Freeman     Bob Gunton   
1  Francis Ford Coppola   Marlon Brando       Al Pacino     James Caan   
2     Christopher Nolan  Christian Bale    Heath Ledger  Aaron Eckhart   
3  Francis Ford Coppola       Al Pacino  Robert De Niro  Robert Duvall   
4          Sidney Lumet     Henry Fonda     Lee J. Cobb  Martin Balsam   

            Star4  No_of_Votes        Gross  
0  William Sadler      2343110   28,341,469  
1    Diane Keaton      1620367  134,966,411  
2   Michael Caine      2303232  534,858,444  
3    Diane Keaton      1129952   57,300,000  
4    John Fiedler       689845    4,360,000

Son 3 satırını getirelim:

In [101]:
df.tail(3)

Poster_Link           Series_Title  \
997  https://m.media-amazon.com/images/M/MV5BM2U3Yz...  From Here to Eternity   
998  https://m.media-amazon.com/images/M/MV5BZTBmMj...               Lifeboat   
999  https://m.media-amazon.com/images/M/MV5BMTY5OD...           The 39 Steps   

    Released_Year Certificate  Runtime                     Genre  IMDB_Rating  \
997          1953      Passed  118 min       Drama, Romance, War          7.6   
998          1944         NaN   97 min                Drama, War          7.6   
999          1935         NaN   86 min  Crime, Mystery, Thriller          7.6   

                                              Overview  Meta_score  \
997  In Hawaii in 1941, a private is cruelly punish...        85.0   
998  Several survivors of a torpedoed merchant ship...        78.0   
999  A man in London tries to help a counter-espion...        93.0   

             Director              Star1              Star2           Star3  \
997    Fred Zinnemann     Burt Lancaster   Montgomery Clift    Deborah Kerr   
998  Alfred Hitchcock  Tallulah Bankhead        John Hodiak   Walter Slezak   
999  Alfred Hitchcock       Robert Donat  Madeleine Carroll  Lucie Mannheim   

              Star4  No_of_Votes       Gross  
997      Donna Reed        43374  30,500,000  
998  William Bendix        26471         NaN  
999  Godfrey Tearle        51853         NaN

Verilerde eksik hücreleri görme:

In [102]:
df.isnull()

Poster_Link  Series_Title  Released_Year  Certificate  Runtime  Genre  \
0          False         False          False        False    False  False   
1          False         False          False        False    False  False   
2          False         False          False        False    False  False   
3          False         False          False        False    False  False   
4          False         False          False        False    False  False   
..           ...           ...            ...          ...      ...    ...   
995        False         False          False        False    False  False   
996        False         False          False        False    False  False   
997        False         False          False        False    False  False   
998        False         False          False         True    False  False   
999        False         False          False         True    False  False   

     IMDB_Rating  Overview  Meta_score  Director  Star1  Star2  Star3  Star4  \
0          False     False       False     False  False  False  False  False   
1          False     False       False     False  False  False  False  False   
2          False     False       False     False  False  False  False  False   
3          False     False       False     False  False  False  False  False   
4          False     False       False     False  False  False  False  False   
..           ...       ...         ...       ...    ...    ...    ...    ...   
995        False     False       False     False  False  False  False  False   
996        False     False       False     False  False  False  False  False   
997        False     False       False     False  False  False  False  False   
998        False     False       False     False  False  False  False  False   
999        False     False       False     False  False  False  False  False   

     No_of_Votes  Gross  
0          False  False  
1          False  False  
2          False  False  
3          False  False  
4          False  False  
..           ...    ...  
995        False   True  
996        False   True  
997        False  False  
998        False   True  
999        False   True  

[1000 rows x 16 columns]

Birden fazla sütun bir değişkene atandığında o değişken de bir dataframe objesi oluyor:

In [103]:
df2=df[['Series_Title', 'IMDB_Rating']]
df2

Series_Title  IMDB_Rating
0    The Shawshank Redemption          9.3
1               The Godfather          9.2
2             The Dark Knight          9.0
3      The Godfather: Part II          9.0
4                12 Angry Men          9.0
..                        ...          ...
995    Breakfast at Tiffany's          7.6
996                     Giant          7.6
997     From Here to Eternity          7.6
998                  Lifeboat          7.6
999              The 39 Steps          7.6

[1000 rows x 2 columns]

Çıkış yılına göre küçükten büyüğe sıralamak için:

In [104]:
df.sort_values('Released_Year')

Poster_Link  \
321  https://m.media-amazon.com/images/M/MV5BNWJiNG...   
127  https://m.media-amazon.com/images/M/MV5BZjhhMT...   
568  https://m.media-amazon.com/images/M/MV5BMTAxYj...   
194  https://m.media-amazon.com/images/M/MV5BZWFhOG...   
193  https://m.media-amazon.com/images/M/MV5BZjEyOT...   
..                                                 ...   
464  https://m.media-amazon.com/images/M/MV5BNmI0MT...   
18   https://m.media-amazon.com/images/M/MV5BNjViNW...   
20   https://m.media-amazon.com/images/M/MV5BOTc2ZT...   
613  https://m.media-amazon.com/images/M/MV5BOTNjM2...   
966  https://m.media-amazon.com/images/M/MV5BNjEzYj...   

                     Series_Title Released_Year Certificate  Runtime  \
321  Das Cabinet des Dr. Caligari          1920         NaN   76 min   
127                       The Kid          1921      Passed   68 min   
568                     Nosferatu          1922         NaN   94 min   
194                  Sherlock Jr.          1924      Passed   45 min   
193                 The Gold Rush          1925      Passed   95 min   
..                            ...           ...         ...      ...   
464                   Dil Bechara          2020          UA  101 min   
18                       Hamilton          2020       PG-13  160 min   
20                Soorarai Pottru          2020           U  153 min   
613                          Druk          2020         NaN  117 min   
966                     Apollo 13            PG           U  140 min   

                         Genre  IMDB_Rating  \
321   Fantasy, Horror, Mystery          8.1   
127      Comedy, Drama, Family          8.3   
568            Fantasy, Horror          7.9   
194    Action, Comedy, Romance          8.2   
193   Adventure, Comedy, Drama          8.2   
..                         ...          ...   
464     Comedy, Drama, Romance          7.9   
18   Biography, Drama, History          8.6   
20                       Drama          8.6   
613              Comedy, Drama          7.8   
966  Adventure, Drama, History          7.6   

                                              Overview  Meta_score  \
321  Hypnotist Dr. Caligari uses a somnambulist, Ce...         NaN   
127  The Tramp cares for an abandoned child, but ev...         NaN   
568  Vampire Count Orlok expresses interest in a ne...         NaN   
194  A film projectionist longs to be a detective, ...         NaN   
193  A prospector goes to the Klondike in search of...         NaN   
..                                                 ...         ...   
464  The emotional journey of two hopelessly in lov...         NaN   
18   The real life of one of America's foremost fou...        90.0   
20   Nedumaaran Rajangam "Maara" sets out to make t...         NaN   
613  Four friends, all high school teachers, test a...        81.0   
966  NASA must devise a strategy to return Apollo 1...        77.0   

              Director                 Star1              Star2  \
321       Robert Wiene         Werner Krauss       Conrad Veidt   
127    Charles Chaplin       Charles Chaplin     Edna Purviance   
568        F.W. Murnau           Max Schreck  Alexander Granach   
194      Buster Keaton         Buster Keaton    Kathryn McGuire   
193    Charles Chaplin       Charles Chaplin         Mack Swain   
..                 ...                   ...                ...   
464     Mukesh Chhabra  Sushant Singh Rajput     Sanjana Sanghi   
18         Thomas Kail    Lin-Manuel Miranda       Phillipa Soo   
20       Sudha Kongara                Suriya           Madhavan   
613  Thomas Vinterberg        Mads Mikkelsen   Thomas Bo Larsen   
966         Ron Howard             Tom Hanks        Bill Paxton   

                     Star3                   Star4  No_of_Votes        Gross  
321        Friedrich Feher             Lil Dagover        57428          NaN  
127          Jackie Coogan             Carl Miller       113314    5,450,000  
568  Gustav von Wangenheim       

Çıkış yılına göre büyükten küçüğe sıralamak için:

In [105]:
df.sort_values(by='Released_Year', ascending=False)

Poster_Link  \
966  https://m.media-amazon.com/images/M/MV5BNjEzYj...   
20   https://m.media-amazon.com/images/M/MV5BOTc2ZT...   
612  https://m.media-amazon.com/images/M/MV5BYjYzOG...   
613  https://m.media-amazon.com/images/M/MV5BOTNjM2...   
18   https://m.media-amazon.com/images/M/MV5BNjViNW...   
..                                                 ...   
193  https://m.media-amazon.com/images/M/MV5BZjEyOT...   
194  https://m.media-amazon.com/images/M/MV5BZWFhOG...   
568  https://m.media-amazon.com/images/M/MV5BMTAxYj...   
127  https://m.media-amazon.com/images/M/MV5BZjhhMT...   
321  https://m.media-amazon.com/images/M/MV5BNWJiNG...   

                     Series_Title Released_Year Certificate  Runtime  \
966                     Apollo 13            PG           U  140 min   
20                Soorarai Pottru          2020           U  153 min   
612    The Trial of the Chicago 7          2020           R  129 min   
613                          Druk          2020         NaN  117 min   
18                       Hamilton          2020       PG-13  160 min   
..                            ...           ...         ...      ...   
193                 The Gold Rush          1925      Passed   95 min   
194                  Sherlock Jr.          1924      Passed   45 min   
568                     Nosferatu          1922         NaN   94 min   
127                       The Kid          1921      Passed   68 min   
321  Das Cabinet des Dr. Caligari          1920         NaN   76 min   

                         Genre  IMDB_Rating  \
966  Adventure, Drama, History          7.6   
20                       Drama          8.6   
612   Drama, History, Thriller          7.8   
613              Comedy, Drama          7.8   
18   Biography, Drama, History          8.6   
..                         ...          ...   
193   Adventure, Comedy, Drama          8.2   
194    Action, Comedy, Romance          8.2   
568            Fantasy, Horror          7.9   
127      Comedy, Drama, Family          8.3   
321   Fantasy, Horror, Mystery          8.1   

                                              Overview  Meta_score  \
966  NASA must devise a strategy to return Apollo 1...        77.0   
20   Nedumaaran Rajangam "Maara" sets out to make t...         NaN   
612  The story of 7 people on trial stemming from v...        77.0   
613  Four friends, all high school teachers, test a...        81.0   
18   The real life of one of America's foremost fou...        90.0   
..                                                 ...         ...   
193  A prospector goes to the Klondike in search of...         NaN   
194  A film projectionist longs to be a detective, ...         NaN   
568  Vampire Count Orlok expresses interest in a ne...         NaN   
127  The Tramp cares for an abandoned child, but ev...         NaN   
321  Hypnotist Dr. Caligari uses a somnambulist, Ce...         NaN   

              Director               Star1              Star2  \
966         Ron Howard           Tom Hanks        Bill Paxton   
20       Sudha Kongara              Suriya           Madhavan   
612       Aaron Sorkin      Eddie Redmayne         Alex Sharp   
613  Thomas Vinterberg      Mads Mikkelsen   Thomas Bo Larsen   
18         Thomas Kail  Lin-Manuel Miranda       Phillipa Soo   
..                 ...                 ...                ...   
193    Charles Chaplin     Charles Chaplin         Mack Swain   
194      Buster Keaton       Buster Keaton    Kathryn McGuire   
568        F.W. Murnau         Max Schreck  Alexander Granach   
127    Charles Chaplin     Charles Chaplin     Edna Purviance   
321       Robert Wiene       Werner Krauss       Conrad Veidt   

                     Star3                   Star4  No_of_Votes        Gross  
966            Kevin Bacon             Gary Sinise       269197  173,837,933  
20            Paresh Rawal       Aparna Balamurali        54995          NaN  
612      Sacha Baron Cohen           Jeremy Strong       

Hangi yılda kaç film çıkmıştır:

In [106]:
df['Released_Year'].value_counts()

2014    32
2004    31
2009    29
2013    28
2016    28
        ..
1926     1
1936     1
1924     1
1921     1
PG       1
Name: Released_Year, Length: 100, dtype: int64

<h3>VERİ FİLTRELEME<h3>

The Dark Knight filminin IMDB Puanını bulalım:

In [107]:
df.loc[df['Series_Title']=="The Dark Knight"]["IMDB_Rating"]

2    9.0
Name: IMDB_Rating, dtype: float64

IMDB'den 7.8'in üstünde olan ve 150000'den fazla oy alan filmleri getirelim:

In [108]:
df.loc[(df['IMDB_Rating']>=7.8) & (df['No_of_Votes']>=150000)]

Poster_Link  \
0    https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1    https://m.media-amazon.com/images/M/MV5BM2MyNj...   
2    https://m.media-amazon.com/images/M/MV5BMTMxNT...   
3    https://m.media-amazon.com/images/M/MV5BMWMwMG...   
4    https://m.media-amazon.com/images/M/MV5BMWU4N2...   
..                                                 ...   
683  https://m.media-amazon.com/images/M/MV5BOTM5N2...   
685  https://m.media-amazon.com/images/M/MV5BMTkxMj...   
688  https://m.media-amazon.com/images/M/MV5BMTQ2OD...   
698  https://m.media-amazon.com/images/M/MV5BZTllND...   
704  https://m.media-amazon.com/images/M/MV5BNmJkOD...   

                            Series_Title Released_Year Certificate  Runtime  \
0               The Shawshank Redemption          1994           A  142 min   
1                          The Godfather          1972           A  175 min   
2                        The Dark Knight          2008          UA  152 min   
3                 The Godfather: Part II          1974           A  202 min   
4                           12 Angry Men          1957           U   96 min   
..                                   ...           ...         ...      ...   
683                   The Breakfast Club          1985          UA   97 min   
685                         Ghostbusters          1984          UA  105 min   
688           E.T. the Extra-Terrestrial          1982           U  115 min   
698  Willy Wonka & the Chocolate Factory          1971           U  100 min   
704                         Mary Poppins          1964           U  139 min   

                        Genre  IMDB_Rating  \
0                       Drama          9.3   
1                Crime, Drama          9.2   
2        Action, Crime, Drama          9.0   
3                Crime, Drama          9.0   
4                Crime, Drama          9.0   
..                        ...          ...   
683             Comedy, Drama          7.8   
685   Action, Comedy, Fantasy          7.8   
688            Family, Sci-Fi          7.8   
698  Family, Fantasy, Musical          7.8   
704   Comedy, Family, Fantasy          7.8   

                                              Overview  Meta_score  \
0    Two imprisoned men bond over a number of years...        80.0   
1    An organized crime dynasty's aging patriarch t...       100.0   
2    When the menace known as the Joker wreaks havo...        84.0   
3    The early life and career of Vito Corleone in ...        90.0   
4    A jury holdout attempts to prevent a miscarria...        96.0   
..                                                 ...         ...   
683  Five high school students meet in Saturday det...        66.0   
685  Three former parapsychology professors set up ...        71.0   
688  A troubled child summons the courage to help a...        91.0   
698  A poor but hopeful boy seeks one of the five c...        67.0   
704  In turn of the century London, a magical nanny...        88.0   

                 Director           Star1           Star2             Star3  \
0          Frank Darabont     Tim Robbins  Morgan Freeman        Bob Gunton   
1    Francis Ford Coppola   Marlon Brando       Al Pacino        James Caan   
2       Christopher Nolan  Christian Bale    Heath Ledger     Aaron Eckhart   
3    Francis Ford Coppola       Al Pacino  Robert De Niro     Robert Duvall   
4            Sidney Lumet     Henry Fonda     Lee J. Cobb     Martin Balsam   
..                    ...             ...             ...               ...   
683           John Hughes  Emilio Estevez     Judd Nelson    Molly Ringwald   
685          Ivan Reitman     Bill Murray     Dan Aykroyd  Sigourney Weaver   
688      Steven Spielberg    Henry Thomas  Drew Barrymore      Peter Coyote   
698            Mel Stuart     Gene Wilder  Jack Albertson      Peter Ostrum   
704      Robert Stevenson   Julie Andrews   Dick Van Dyke   David Tomlinson   

              Star4  No_of_Votes        Gross  
0    W

DataFrame kopyalama

In [109]:
df_copy=df.copy()
df_copy.head()

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   
2  https://m.media-amazon.com/images/M/MV5BMTMxNT...   
3  https://m.media-amazon.com/images/M/MV5BMWMwMG...   
4  https://m.media-amazon.com/images/M/MV5BMWU4N2...   

               Series_Title Released_Year Certificate  Runtime  \
0  The Shawshank Redemption          1994           A  142 min   
1             The Godfather          1972           A  175 min   
2           The Dark Knight          2008          UA  152 min   
3    The Godfather: Part II          1974           A  202 min   
4              12 Angry Men          1957           U   96 min   

                  Genre  IMDB_Rating  \
0                 Drama          9.3   
1          Crime, Drama          9.2   
2  Action, Crime, Drama          9.0   
3          Crime, Drama          9.0   
4          Crime, Drama          9.0   

                                            Overview  Meta_score  \
0  Two imprisoned men bond over a number of years...        80.0   
1  An organized crime dynasty's aging patriarch t...       100.0   
2  When the menace known as the Joker wreaks havo...        84.0   
3  The early life and career of Vito Corleone in ...        90.0   
4  A jury holdout attempts to prevent a miscarria...        96.0   

               Director           Star1           Star2          Star3  \
0        Frank Darabont     Tim Robbins  Morgan Freeman     Bob Gunton   
1  Francis Ford Coppola   Marlon Brando       Al Pacino     James Caan   
2     Christopher Nolan  Christian Bale    Heath Ledger  Aaron Eckhart   
3  Francis Ford Coppola       Al Pacino  Robert De Niro  Robert Duvall   
4          Sidney Lumet     Henry Fonda     Lee J. Cobb  Martin Balsam   

            Star4  No_of_Votes        Gross  
0  William Sadler      2343110   28,341,469  
1    Diane Keaton      1620367  134,966,411  
2   Michael Caine      2303232  534,858,444  
3    Diane Keaton      1129952   57,300,000  
4    John Fiedler       689845    4,360,000

Row Drop (satır silme)

In [110]:
df_new2=df.drop([2])
df_new2.head()

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   
3  https://m.media-amazon.com/images/M/MV5BMWMwMG...   
4  https://m.media-amazon.com/images/M/MV5BMWU4N2...   
5  https://m.media-amazon.com/images/M/MV5BNzA5ZD...   

                                    Series_Title Released_Year Certificate  \
0                       The Shawshank Redemption          1994           A   
1                                  The Godfather          1972           A   
3                         The Godfather: Part II          1974           A   
4                                   12 Angry Men          1957           U   
5  The Lord of the Rings: The Return of the King          2003           U   

   Runtime                     Genre  IMDB_Rating  \
0  142 min                     Drama          9.3   
1  175 min              Crime, Drama          9.2   
3  202 min              Crime, Drama          9.0   
4   96 min              Crime, Drama          9.0   
5  201 min  Action, Adventure, Drama          8.9   

                                            Overview  Meta_score  \
0  Two imprisoned men bond over a number of years...        80.0   
1  An organized crime dynasty's aging patriarch t...       100.0   
3  The early life and career of Vito Corleone in ...        90.0   
4  A jury holdout attempts to prevent a miscarria...        96.0   
5  Gandalf and Aragorn lead the World of Men agai...        94.0   

               Director          Star1            Star2          Star3  \
0        Frank Darabont    Tim Robbins   Morgan Freeman     Bob Gunton   
1  Francis Ford Coppola  Marlon Brando        Al Pacino     James Caan   
3  Francis Ford Coppola      Al Pacino   Robert De Niro  Robert Duvall   
4          Sidney Lumet    Henry Fonda      Lee J. Cobb  Martin Balsam   
5         Peter Jackson    Elijah Wood  Viggo Mortensen   Ian McKellen   

            Star4  No_of_Votes        Gross  
0  William Sadler      2343110   28,341,469  
1    Diane Keaton      1620367  134,966,411  
3    Diane Keaton      1129952   57,300,000  
4    John Fiedler       689845    4,360,000  
5   Orlando Bloom      1642758  377,845,905

<h3>Filtreleme<h3>

In [111]:
df_filtered=df[df['IMDB_Rating']>=9.1]
df_filtered

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   

               Series_Title Released_Year Certificate  Runtime         Genre  \
0  The Shawshank Redemption          1994           A  142 min         Drama   
1             The Godfather          1972           A  175 min  Crime, Drama   

   IMDB_Rating                                           Overview  Meta_score  \
0          9.3  Two imprisoned men bond over a number of years...        80.0   
1          9.2  An organized crime dynasty's aging patriarch t...       100.0   

               Director          Star1           Star2       Star3  \
0        Frank Darabont    Tim Robbins  Morgan Freeman  Bob Gunton   
1  Francis Ford Coppola  Marlon Brando       Al Pacino  James Caan   

            Star4  No_of_Votes        Gross  
0  William Sadler      2343110   28,341,469  
1    Diane Keaton      1620367  134,966,411

In [112]:
df_filtered=df.query('IMDB_Rating>=9.1')
df_filtered

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   

               Series_Title Released_Year Certificate  Runtime         Genre  \
0  The Shawshank Redemption          1994           A  142 min         Drama   
1             The Godfather          1972           A  175 min  Crime, Drama   

   IMDB_Rating                                           Overview  Meta_score  \
0          9.3  Two imprisoned men bond over a number of years...        80.0   
1          9.2  An organized crime dynasty's aging patriarch t...       100.0   

               Director          Star1           Star2       Star3  \
0        Frank Darabont    Tim Robbins  Morgan Freeman  Bob Gunton   
1  Francis Ford Coppola  Marlon Brando       Al Pacino  James Caan   

            Star4  No_of_Votes        Gross  
0  William Sadler      2343110   28,341,469  
1    Diane Keaton      1620367  134,966,411

runtime veri tipine bakalım

In [113]:
df.dtypes

Poster_Link       object
Series_Title      object
Released_Year     object
Certificate       object
Runtime           object
Genre             object
IMDB_Rating      float64
Overview          object
Meta_score       float64
Director          object
Star1             object
Star2             object
Star3             object
Star4             object
No_of_Votes        int64
Gross             object
dtype: object

Runtime içinde 140 min gibi veri kullanıldığı için bölüyoruz int ve string olarak

In [114]:
f=lambda x:(x["Runtime"].split())[0]
df["Runtime2"]=df.apply(f, axis=1)
df.head()

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   
2  https://m.media-amazon.com/images/M/MV5BMTMxNT...   
3  https://m.media-amazon.com/images/M/MV5BMWMwMG...   
4  https://m.media-amazon.com/images/M/MV5BMWU4N2...   

               Series_Title Released_Year Certificate  Runtime  \
0  The Shawshank Redemption          1994           A  142 min   
1             The Godfather          1972           A  175 min   
2           The Dark Knight          2008          UA  152 min   
3    The Godfather: Part II          1974           A  202 min   
4              12 Angry Men          1957           U   96 min   

                  Genre  IMDB_Rating  \
0                 Drama          9.3   
1          Crime, Drama          9.2   
2  Action, Crime, Drama          9.0   
3          Crime, Drama          9.0   
4          Crime, Drama          9.0   

                                            Overview  Meta_score  \
0  Two imprisoned men bond over a number of years...        80.0   
1  An organized crime dynasty's aging patriarch t...       100.0   
2  When the menace known as the Joker wreaks havo...        84.0   
3  The early life and career of Vito Corleone in ...        90.0   
4  A jury holdout attempts to prevent a miscarria...        96.0   

               Director           Star1           Star2          Star3  \
0        Frank Darabont     Tim Robbins  Morgan Freeman     Bob Gunton   
1  Francis Ford Coppola   Marlon Brando       Al Pacino     James Caan   
2     Christopher Nolan  Christian Bale    Heath Ledger  Aaron Eckhart   
3  Francis Ford Coppola       Al Pacino  Robert De Niro  Robert Duvall   
4          Sidney Lumet     Henry Fonda     Lee J. Cobb  Martin Balsam   

            Star4  No_of_Votes        Gross Runtime2  
0  William Sadler      2343110   28,341,469      142  
1    Diane Keaton      1620367  134,966,411      175  
2   Michael Caine      2303232  534,858,444      152  
3    Diane Keaton      1129952   57,300,000      202  
4    John Fiedler       689845    4,360,000       96

Runtime2 olarak oluşturduğumuz sutunun veri tipine bakalım

In [115]:
df.dtypes

Poster_Link       object
Series_Title      object
Released_Year     object
Certificate       object
Runtime           object
Genre             object
IMDB_Rating      float64
Overview          object
Meta_score       float64
Director          object
Star1             object
Star2             object
Star3             object
Star4             object
No_of_Votes        int64
Gross             object
Runtime2          object
dtype: object

runtime2 sutunu obje olarak görülüyor bunu integer olarak değiştireğiz ve 130 dakikadan fazla olan filmlerini listeliycez

In [116]:
df['Runtime2']=df['Runtime2'].astype('int')
df_filtered=df.query("Runtime2 >= 130")
df_filtered

Poster_Link  \
0    https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1    https://m.media-amazon.com/images/M/MV5BM2MyNj...   
2    https://m.media-amazon.com/images/M/MV5BMTMxNT...   
3    https://m.media-amazon.com/images/M/MV5BMWMwMG...   
5    https://m.media-amazon.com/images/M/MV5BNzA5ZD...   
..                                                 ...   
974  https://m.media-amazon.com/images/M/MV5BNWFlYW...   
988  https://m.media-amazon.com/images/M/MV5BMjM1Nj...   
990  https://m.media-amazon.com/images/M/MV5BYjRmY2...   
991  https://m.media-amazon.com/images/M/MV5BMzAyND...   
996  https://m.media-amazon.com/images/M/MV5BODk3Yj...   

                                      Series_Title Released_Year Certificate  \
0                         The Shawshank Redemption          1994           A   
1                                    The Godfather          1972           A   
2                                  The Dark Knight          2008          UA   
3                           The Godfather: Part II          1974           A   
5    The Lord of the Rings: The Return of the King          2003           U   
..                                             ...           ...         ...   
974                        The Godfather: Part III          1990           A   
988             Close Encounters of the Third Kind          1977           U   
990                                   Giù la testa          1971          PG   
991                                 Kelly's Heroes          1970          GP   
996                                          Giant          1956           G   

     Runtime                     Genre  IMDB_Rating  \
0    142 min                     Drama          9.3   
1    175 min              Crime, Drama          9.2   
2    152 min      Action, Crime, Drama          9.0   
3    202 min              Crime, Drama          9.0   
5    201 min  Action, Adventure, Drama          8.9   
..       ...                       ...          ...   
974  162 min              Crime, Drama          7.6   
988  138 min             Drama, Sci-Fi          7.6   
990  157 min       Drama, War, Western          7.6   
991  144 min    Adventure, Comedy, War          7.6   
996  201 min            Drama, Western          7.6   

                                              Overview  Meta_score  \
0    Two imprisoned men bond over a number of years...        80.0   
1    An organized crime dynasty's aging patriarch t...       100.0   
2    When the menace known as the Joker wreaks havo...        84.0   
3    The early life and career of Vito Corleone in ...        90.0   
5    Gandalf and Aragorn lead the World of Men agai...        94.0   
..                                                 ...         ...   
974  Follows Michael Corleone, now in his 60s, as h...        60.0   
988  Roy Neary, an electric lineman, watches how hi...        90.0   
990  A low-life bandit and an I.R.A. explosives exp...        77.0   
991  A group of U.S. soldiers sneaks across enemy l...        50.0   
996  Sprawling epic covering the life of a Texas ca...        84.0   

                 Director             Star1              Star2          Star3  \
0          Frank Darabont       Tim Robbins     Morgan Freeman     Bob Gunton   
1    Francis Ford Coppola     Marlon Brando          Al Pacino     James Caan   
2       Christopher Nolan    Christian Bale       Heath Ledger  Aaron Eckhart   
3    Francis Ford Coppola         Al Pacino     Robert De Niro  Robert Duvall   
5           Peter Jackson       Elijah Wood    Viggo Mortensen   Ian McKellen   
..                    ...               ...                ...            ...   
974  Francis Ford Coppola         Al Pacino       Diane Keaton    Andy Garcia   
988      Steven Spielberg  Richard Dreyfuss  François Truffaut      Teri Garr   
990          Sergio Leone       Rod Steiger       James Coburn   Romolo Valli   
991       Brian G. Hutton    Clint Eastwood      Telly Savalas    Don Ric

Yeni oluşturduğum Runtime2 sutununu silebiliriz artık.

In [117]:
df = df.drop(['Runtime2'], axis=1)
df.head()

Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   
2  https://m.media-amazon.com/images/M/MV5BMTMxNT...   
3  https://m.media-amazon.com/images/M/MV5BMWMwMG...   
4  https://m.media-amazon.com/images/M/MV5BMWU4N2...   

               Series_Title Released_Year Certificate  Runtime  \
0  The Shawshank Redemption          1994           A  142 min   
1             The Godfather          1972           A  175 min   
2           The Dark Knight          2008          UA  152 min   
3    The Godfather: Part II          1974           A  202 min   
4              12 Angry Men          1957           U   96 min   

                  Genre  IMDB_Rating  \
0                 Drama          9.3   
1          Crime, Drama          9.2   
2  Action, Crime, Drama          9.0   
3          Crime, Drama          9.0   
4          Crime, Drama          9.0   

                                            Overview  Meta_score  \
0  Two imprisoned men bond over a number of years...        80.0   
1  An organized crime dynasty's aging patriarch t...       100.0   
2  When the menace known as the Joker wreaks havo...        84.0   
3  The early life and career of Vito Corleone in ...        90.0   
4  A jury holdout attempts to prevent a miscarria...        96.0   

               Director           Star1           Star2          Star3  \
0        Frank Darabont     Tim Robbins  Morgan Freeman     Bob Gunton   
1  Francis Ford Coppola   Marlon Brando       Al Pacino     James Caan   
2     Christopher Nolan  Christian Bale    Heath Ledger  Aaron Eckhart   
3  Francis Ford Coppola       Al Pacino  Robert De Niro  Robert Duvall   
4          Sidney Lumet     Henry Fonda     Lee J. Cobb  Martin Balsam   

            Star4  No_of_Votes        Gross  
0  William Sadler      2343110   28,341,469  
1    Diane Keaton      1620367  134,966,411  
2   Michael Caine      2303232  534,858,444  
3    Diane Keaton      1129952   57,300,000  
4    John Fiedler       689845    4,360,000